frontend/src/modules/report/FeatureListContainer/FeatureFilterButtons.tsx

```diff
@@ -2,54 +2,64 @@ import React, { FC, ReactNode } from 'react';
 import { Card, IconButton, Tooltip } from '@material-ui/core';
 import { CheckCircleOutline, ErrorOutline, RemoveCircleOutline, HelpOutline } from '@material-ui/icons';
 import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
 
-import Status, { StatusMap, Passed, Failed, Skipped, Undefined, Statuses } from 'models/Status';
+import Status, { StatusMap, Passed, Failed, Skipped, Undefined, Statuses, SelectedStatuses } from 'models/Status';
 import { useFeatureFilterButtonStyles } from './styles/FeatureListContainerStyles';
 import { useStatusColorStyles } from 'modules/styles/globalStyles';
-
-interface Props {
-  selectedStatus: StatusMap<boolean>;
-  handleFilterButtonClick(status: Status): void;
-}
+import { RootStore } from 'rootReducer';
 
 interface FilterButtonProps {
   status: Status;
-  selectedStatus: StatusMap<boolean>;
-  handleFilterButtonClick(status: Status): void;
 }
 
-const FilterButton: FC<FilterButtonProps> = ({ status, selectedStatus, handleFilterButtonClick }) => {
+const generateQueryParams = (selectedStatuses: SelectedStatuses, status: Status): string => {
+  let queryParams = '';
+  Object.entries(selectedStatuses).forEach(([key, val]) => {
+    if (key === status && val) {
+      queryParams += `hidden=${key}&`;
```

Comment:
There's a library to do this query params stuff more easily which we've used elsewhere, or alternatively can use an array and join.